### PR TITLE
Remove 'add to cart' button for logged-in restaurant owners

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT Story: https://www.pivotaltracker.com/story/show/131445807

Changes proposed in this pull request:
- Logged in restaurant owners cannot see add to cart button (they're not allowed to have a shopping cart - so it throws an error)

What I have learned working on this feature: I probably hadn't used an "unless" statement before, actually.

Ready for review @tochman !

### CODE REVIEW

File: app/views/dishes/show.html.haml [hunk 1]
```diff
@@ -14,4 +14,5 @@ Menu(s):
 - @dish.menus.each do |menu|
   = menu.title
 %br/
-= link_to 'Add to cart', dish_add_item_path(@dish), method: :post, class: "btn btn-success"
+- unless !current_user.nil? && current_user.role == 'owner'
```
Comment (tochman) [inline app/views/dishes/show.html.haml]: Use `current_user.owner?`

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/views/dishes/show.html.haml (hunk 1 above)
Comment (AmberWilkie) [reply app/views/dishes/show.html.haml]: yes makes sense. refactored.

Comment (tochman) [reply app/views/dishes/show.html.haml]: Or
If current_user && current_user.custoner?

Comment (tochman) [reply app/views/dishes/show.html.haml]: Or use an ability;-)

Comment (AmberWilkie) [reply app/views/dishes/show.html.haml]: Nah cuz visitors can add things to their cart. Will have to make it `current_user.owner? || current_user.admin?` when we get around to adding admins, but everybody else should see that button

Comment (tochman) [reply app/views/dishes/show.html.haml]: I don't agree ;-) Pushed up a commit that uses ability (we can revert if that is not a solution you approve)

**Review by tochman — APPROVED** [review]

**Comment by AmberWilkie** [comment]
Ok I dig it. I guess I was thinking about authorization the wrong way.

**Comment by tochman** [comment]
Cool

**Pull request merged**